src/compas/geometry/primitives/shapes/capsule.py

```diff
@@ -0,0 +1,228 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from math import cos
+from math import pi
+from math import sin
+
+from compas.geometry import matrix_from_frame
+from compas.geometry import transform_points
+from compas.geometry import Frame
+from compas.geometry import Plane
+from compas.geometry.primitives import Line
+from compas.geometry.primitives.shapes import Shape
+
+__all__ = ['Capsule']
+
+
+class Capsule(Shape):
+    """A capsule is defined by a line segment and a radius.
+
+    Attributes
+    ----------
+    segment: :class:`compas.geometry.Line`
+        The axis line of the capsule.
+    radius: float
+        The radius of the capsule.
+
+    Examples
+    --------
+    >>> TODO
+
+    """
+
+    __slots__ = ['_segment', '_radius']
+
+    def __init__(self, line, radius):
+        self._line = None
+        self._radius = None
+        self.line = line
+        self.radius = radius
+
+    @property
+    def line(self):
+        return self._line
+
+    @line.setter
+    def line(self, line):
+        self._line = line
+
+    @property
+    def start(self):
+        return self.line.start
+
+    @property
+    def end(self):
+        return self.line.end
+
+    @property
+    def radius(self):
+        """float: The radius of the capsule."""
+        return self._radius
+
+    @radius.setter
+    def radius(self, radius):
+        self._radius = float(abs(radius))
+
+    @property
+    def length(self):
+        self.line.length
+
+    @property
+    def volume(self):
+        # cylinder plus 2 half spheres
+        cylinder = self.radius**2 * pi * self.length
+        caps = 4./3. * pi * self.radius**3
+        return cylinder + caps
+
+    @property
+    def area(self):
+        # cylinder minus caps plus 2 half spheres
+        cylinder = self.radius*2 * pi * self.length
+        caps = 4 * pi * self.radius**2
+        return cylinder + caps
+
+    @property
+    def data(self):
+        """Returns the data dictionary that represents the capsule.
+
+        Returns
+        -------
+        dict
+            The capsule data.
+
+        """
+        return {'line': self.line.data,
+                'radius': self.radius}
+
+    @data.setter
+    def data(self, data):
+        self.line = Line.from_data(data['line'])
+        self.radius = data['radius']
+
+    def to_data(self):
+        """Returns the data dictionary that represents the capsule.
+
+        Returns
+        -------
+        dict
+            The capsule data.
+
+        """
+        return self.data
+
```

Comment:
what about a `__repr__` that returns a string that can be used to construct a Capsule again, like:
```suggestion
    def __repr__(self):
        return "Capsule({0}, {1})".format(self.line, self.radius)
```

Reply:
does `repr` get applied recursively? i mean, will this apply `repr` to `self.line` as well?